packages/core/src/metrics/instance.ts

```diff
@@ -0,0 +1,106 @@
+import type { MetricInstance } from '@sentry/types';
+import { COUNTER_METRIC_TYPE, DISTRIBUTION_METRIC_TYPE, GAUGE_METRIC_TYPE, SET_METRIC_TYPE } from './constants';
+import { simpleHash } from './utils';
+
+/**
+ * A metric instance representing a counter.
+ */
+export class CounterMetric implements MetricInstance {
+  public constructor(private _value: number) {}
+
+  /** @inheritdoc */
+  public add(value: number): void {
+    this._value += value;
+  }
+
+  /** @inheritdoc */
+  public toString(): string {
+    return `${this._value}`;
+  }
+}
+
+/**
+ * A metric instance representing a gauge.
+ */
+export class GaugeMetric implements MetricInstance {
+  private _last: number;
+  private _min: number;
+  private _max: number;
+  private _sum: number;
+  private _count: number;
+
+  public constructor(value: number) {
+    this._last = value;
+    this._min = value;
+    this._max = value;
+    this._sum = value;
+    this._count = 1;
+  }
+
+  /** @inheritdoc */
+  public add(value: number): void {
+    this._last = value;
+    this._min = Math.min(this._min, value);
+    this._max = Math.max(this._max, value);
```

Comment:
Instead of calling Math.min and later Math.max, a single if statement to understand which is smaller/bigger, would be a lot of faster.